cmd/cloud-endpoints-controller/main.go

```diff
@@ -347,11 +318,74 @@ func changeDetected(parent *CloudEndpoint, children *CloudEndpointControllerRequ
 				}
 			}
 		}
+
+		if parent.Spec.OpenAPISpecConfigMap.Name != "" {
+			specData, err := getConfigMapSpecData(parent.ObjectMeta.Namespace, parent.Spec.OpenAPISpecConfigMap.Name, parent.Spec.OpenAPISpecConfigMap.Key)
+			if err != nil || toSha1(specData) != status.ConfigMapHash {
+				log.Printf("[DEBUG][%s] Changed because configmap spec changed.", parent.Name)
+				changed = true
+			}
+		}
 	}
 
 	return changed
 }
 
+func getTargetIngress(parent *CloudEndpoint) (string, []string, error) {
+	var target string
+	var jwtAudiences []string
+
+	ingress, err := config.clientset.ExtensionsV1beta1().Ingresses(parent.Spec.TargetIngress.Namespace).Get(parent.Spec.TargetIngress.Name, metav1.GetOptions{})
+	if err != nil {
+		log.Printf("[INFO][%s] waiting for Ingress %s", parent.Name, parent.Spec.TargetIngress.Name)
+		return "", nil, err
```

Comment:
This should be returning a nil error, not the actual error because waiting for ingress is a nominal state, not a fatal error.

This func should only return a non-nill error if something off-nominal happened.